packages/workflow/src/internals.ts

```diff
@@ -394,10 +399,11 @@ export class Activator implements ActivationHandler {
     }
   }
 
-  getAndResetCommands(): coresdk.workflow_commands.IWorkflowCommand[] {
-    const commands = this.commands;
-    this.commands = [];
-    return commands;
+  concludeActivation(): ActivationCompletion {
+    const commands = this.commands.splice(0);
+    const usedInternalFlags = [...this.newlyUsedFlags];
```

Comment:
Do you really need to distinguish the newly used ones from the rest? Once a flag is on, it's on for good.